File: backend/schemas/retrieval_lab.py
from pydantic import BaseModel, Field

from schemas.chat import RetrievedSource


class RetrievalCompareRequest(BaseModel):
    message: str = Field(min_length=1, max_length=4000)
    top_k: int = Field(default=5, ge=1, le=20)


class RetrievalModeResult(BaseModel):
    mode_id: str
    label: str
    retrieval_mode: str
    retrieval_ms: float
    sources: list[RetrievedSource]


class RetrievalCompareResponse(BaseModel):
    repository_id: int
    question: str
    top_k: int
    results: list[RetrievalModeResult]

File: frontend/src/types/retrievalLab.ts
import type { RetrievedSource } from '@/types/chat'

export type RetrievalCompareRequest = {
  message: string
  top_k?: number
}

export type RetrievalModeResult = {
  mode_id: string
  label: string
  retrieval_mode: 'hybrid' | 'vector'
  retrieval_ms: number
  sources: RetrievedSource[]
}

export type RetrievalCompareResponse = {
  repository_id: number
  question: string
  top_k: number
  results: RetrievalModeResult[]
}
